source/_components/octoprint.markdown

```diff
@@ -14,23 +14,99 @@ ha_release: 0.19
 ha_iot_class: "Local Polling"
 ---
 
-[OctoPrint](http://octoprint.org/) is a web interface for your 3D printer. This is the main component to integrate OctoPrint sensors, you will have to setup sensors and binary sensors separately.
+[OctoPrint](http://octoprint.org/) is a web interface for your 3D printer. This is the main component to integrate OctoPrint sensors.
 
-To get started with the OctoPrint API, please follow the directions on their [site](http://docs.octoprint.org/en/master/api/general.html). Once OctoPrint is configured you will need to add your API key and host to your `configuration.yaml`. 
+## {% linkable_title Base Configuration %}
 
+To get started with the OctoPrint API, please follow the directions on their [site](http://docs.octoprint.org/en/master/api/general.html). Once OctoPrint is configured you will need to add your API key and host to your `configuration.yaml`.
 
 ```yaml
 octoprint:
   host: YOUR_OCTOPRINT_HOST
   api_key: YOUR_API_KEY
-  bed: false
-  number_of_tools: 1
 ```
 
-Configuration variables:
+{% configuration %}
+host:
+  description: IP address or hostname of Octoprint host.
+  required: true
+  type: string
+api_key:
+  description: The retrieved api key.
+  required: true
+  type: string
+name:
+  description: The name for this printer, must be unique if multiple printers are defined.
+  required: false
+  type: string
+  default: 'OctoPrint'
+port:
+  description: The port of the Octoprint server.
+  required: false
+  type: int
+  default: 80
+ssl:
+  description: Enable or disable SSL
+  required: false
+  type: boolean
+  default: false
+bed:
+  description: If the printer has a heated bed.
+  required: false
+  type: boolean
+  default: false
+number_of_tools:
+  description: Number of temperature adjustable tools. i.e. nozzle.
+  required: false
+  type: int
+  default: 1
+sensors:
+  description: Configuration for the sensors
+  required: false
+  type: map
+  keys:
+    monitored_conditions:
+        description: The sensors to activate
+        type: list
+        default: all (`Current State`, `Temperatures`, `Job Percentage`, `Time Elapsed`, `Time Remaining`)
+binary_sensors:
+  description: Configuration for the binary sensors
+  required: false
+  type: map
+  keys:
+    monitored_conditions:
+        description: The sensors to activate
+        type: list
+        default: all (`Printing`, `Printing Error`)
+{% endconfiguration %}
 
-- **host** (*Required*): IP address or hostname of Octoprint host.
-- **api_key** (*Required*): The retrieved api key.
-- **bed** (*Optional*): If the printer has a heated bed.
-- **number_of_tools** (*Optional*): Number of temperature adjustable tools. i.e. nozzle.
+<p class='note'>
+Description of monitored conditions:
 
+  - **Current State**: Text of current state.
```

Comment:
These should be added to the `monitored_condition` as keys in the configuration variables above.

Reply:
Thanks, done.